src/api.v2/controllers/experimentController.js

```diff
@@ -1,41 +1,101 @@
 const userAccess = require('../model/userAccess');
 
 const AccessRole = require('../../utils/enums/AccessRole');
-
-const { OK } = require('../../utils/responses');
-
 const getLogger = require('../../utils/getLogger');
+const { OK } = require('../../utils/responses');
+const { NotFoundError } = require('../../utils/responses');
 
 const logger = getLogger('[ExperimentController] - ');
 
+const getExperiment = async (req, res) => {
+  const { params: { experimentId } } = req;
+
+  const data = await experiment.getExperimentData(experimentId);
+
+  if (_.isNil(data)) {
+    throw new NotFoundError('Experiment not found');
+  }
+
+  res.json(data);
+};
+
 const createExperiment = async (req, res) => {
   const { params: { experimentId }, user, body } = req;
 
   const { name, description } = body;
 
   logger.log('Creating experiment');
 
-  await experiment.create({ id: experimentId, name, description });
+  const experimentCreateResult = await experiment.create({ id: experimentId, name, description });
+
+  if (experimentCreateResult.length === 0) {
+    throw new Error(`Experiment ${experimentId} creation failed`);
+  }
 
   logger.log('Setting up access permissions for experiment');
-  await Promise.all([
+  const userAccessCreateResults = await Promise.all([
     userAccess.create(
       { user_id: user.sub, experiment_id: experimentId, access_role: AccessRole.OWNER },
     ),
     userAccess.create(
-      { user_id: config.adminSub, experiment_id: experimentId, access_role: AccessRole.OWNER },
+      { user_id: config.adminSub, experiment_id: experimentId, access_role: AccessRole.ADMIN },
     ),
   ]);
 
-  logger.log('Finished creating experiment');
+  if (userAccessCreateResults[0].length === 0 || userAccessCreateResults[1].length === 0) {
+    throw new Error(`User access creation failed for experiment ${experimentId}`);
+  }
+
+  logger.log(`Finished creating experiment ${experimentId}`);
+
+  res.json(OK());
+};
+
+const patchExperiment = async (req, res) => {
+  const { params: { experimentId }, body } = req;
+
+  logger.log(`Updating experiment ${experimentId}`);
+
+  const snakeCasedKeysToPatch = _.mapKeys(body, (_value, key) => _.snakeCase(key));
+
+  await experiment.update(experimentId, snakeCasedKeysToPatch);
+
+  logger.log(`Finished updating experiment ${experimentId}`);
+
+  res.json(OK());
+};
+
+const updateSamplePosition = async (req, res) => {
+  const {
+    params: { experimentId },
+    body: { newPosition, oldPosition },
+  } = req;
+
+  logger.log(`Reordering sample in ${experimentId} from position ${oldPosition} to ${newPosition}`);
+
+  if (oldPosition === newPosition) {
+    logger.log('Skipping reordering, oldPosition === newPosition');
+    res.json(OK());
+
+    return;
+  }
+
+  await experiment.updateSamplePosition(experimentId, oldPosition, newPosition);
+
+  logger.log(`Finished reordering samples in ${experimentId}`);
 
   res.json(OK());
 };
 
 module.exports = {
```

Comment:
you mean wrapping everything in a class or something and exporting that? why would it be better?

Reply:
yes, reasons that come to mind are:
1. we were already doing that with all the experiment, projects, etc... services
2. naming conventions demand the file to be named as the exports (in this case it's not possible because there are many different functions.
3. when using this functions from the outside I think it's easier to "understand" them if they are grouped under the same import like `experimentController.getExperiment` & `experimentController.patchExperiment`. 

what would be the cons of doing it this way, and the pros of doing it as you did?

Reply:
1. In the experiment, projects, etc services we were using classes without any need, no parameters were taken and all they did was hold the functions. There were other things like the naming (with underscores instead of camelcase) which we didn't like either and were changed.
2. I don't understand this, they are many functions wrapped in an object, we are exporting the object
3. We can still do that, when you do `require()` you can just do 
`const experimentController = require('../controllers/experimentController');`
instead of 
```
const {
  createExperiment, getExperiment, patchExperiment, updateSamplePosition,
} = require('../controllers/experimentController');
```

The cons is that I don't see why we would use a class when it doesn't add any value.
The pros is that it's the way we have been handling export and import for files that hold many objects in most places in the UI and the api (the only exception I can think of are the `*-services`).